src/classes/OPP_OpportunityContactRoles_TDTM.cls

```diff
@@ -477,93 +865,147 @@ public without sharing class OPP_OpportunityContactRoles_TDTM extends TDTM_Runna
+                    uniqueOCRKeys.remove(calcUniqueOCRKey(primaryOCR));
+                    OpportunityContactRole newPrimaryOCR = getOCR(opp, FIELD_NAME_PRIMARY_CONTACT, isIndividualAccount);
+                    dmlWrapper.objectsToInsert.add(newPrimaryOCR);
+                    transactionOCRs = populateTransactionOCRs(transactionOCRs, newPrimaryOCR);
+                    uniqueOCRKeys.add(calcUniqueOCRKey(newPrimaryOCR));
                 }
             }
         }
     }
 
-
     /*******************************************************************************************************
     * @description Manages create and update of the Opportunity Contact Roles for the Honoree and Notification
     * Recipient contact lookup fields.
-    * @param opp The current opportunity from trigger.new
-    * @param oldOpp The old opportunity from trigger.old
+    * @param obj The current opportunity from trigger.new
+    * @param oldObj The old opportunity from trigger.old
     * @param mapConIdOCR A map of contact id to Opportunity Contact Role records for this opportunity.
     * @param fieldName The API name of the field to verify
+    * @param uniqueOCRKeys A Set of OpportunityId-ContactId-Role values to be checked and augmented to help prevent duplicate roles from being added
     * @param transactionOCRs Contains the OCRs that will be evaluated to ensure compliance with the OCR hierarchy.
     * @param ocrsToDeleteImmediately Holds Opportunity Contact Roles that need to be deleted immediately.
     * @return void This method adds records to dmlWrapper and returns nothing.
     ********************************************************************************************************/
-    private void manageOtherOCR(Opportunity opp, Opportunity oldOpp, Map<Id, OpportunityContactRole> mapConIdOCR, String fieldName,
+    private void manageOtherOCR(SObject obj, SObject oldObj, Map<Id, OpportunityContactRole> mapConIdOCR, String fieldName, Set<String> uniqueOCRKeys,
                                 Map<Id, List<OpportunityContactRole>> transactionOCRs, List<OpportunityContactRole> ocrsToDeleteImmediately) {
-        Id idNewCon = (Id)opp.get(fieldName);
-        Id idOldCon = (Id)oldOpp.get(fieldName);
+        Id idNewCon = (Id) obj.get(fieldName);
+        Id idOldCon = (oldObj != null) ? (Id) oldObj.get(fieldName) : null;
+
+        if (mapConIdOCR == null) // Doing just so repeated null checks are not necessary...
+            mapConIdOCR = new Map<Id, OpportunityContactRole>();
+
         OpportunityContactRole ocrNewCon = mapConIdOCR.get(idNewCon);
         OpportunityContactRole ocrOldCon = mapConIdOCR.get(idOldCon);
 
         if (idNewCon == null) {
             //if the lookup field was blanked out, and an OCR matches the previous contact with the correct role, delete it
-            if (idOldCon!=null && ocrOldCon!=null) {
+            if (idOldCon != null && ocrOldCon != null) {
```

Comment:
If you look closer, you will see that the second and third levels have else logic if that condition is not true.  I could restructure it, but it is not as simple as collapsing, and would rather not (especially since the plan is/should be to replace this code line with a much simpler one).